### PR TITLE
Create filter for TEC status

### PR DESCRIPTION
**Pull request by patriciaahuang**
### Summary

The admin page where the C&I lead can approve/reject TECs previously showed all submission statuses on one page, which caused inconvenient scrolling when there were many submissions. There is now a dropdown menu that can filter through each status and also go back to showing all statuses, if needed.

### Notion/Figma Link

https://www.notion.so/Admin-TEC-Filter-by-status-04f238712c8245689253cfc53b8d065c?pvs=4

### Test Plan

I checked that only TEC submissions of the selected status are shown, the "View All" button only appears when there is a status selected, and approving/rejecting a request will stay on the same selected status when the process is complete.

Screenshot 2024-02-27 at 12 30 44 PM

Screenshot 2024-02-27 at 12 39 36 PM

### CODE REVIEW

File: frontend/src/components/Admin/TeamEvent/TeamEventDetails.module.css
Comment (andrew032011) [inline frontend/src/components/Admin/TeamEvent/TeamEventDetails.module.css:57]: Not sure if `listContainerAll` is being used

File: frontend/src/components/Admin/TeamEvent/TeamEventDetails.module.css
Comment (andrew032011) [inline frontend/src/components/Admin/TeamEvent/TeamEventDetails.module.css:70]: nit: camelCase this

File: frontend/src/components/Admin/TeamEvent/TeamEventDetails.tsx
Comment (andrew032011) [inline frontend/src/components/Admin/TeamEvent/TeamEventDetails.tsx:67]: allStatus should factored out as a constant, placed in `constants.js` file

File: frontend/src/components/Admin/TeamEvent/TeamEventDetails.tsx
Comment (andrew032011) [inline frontend/src/components/Admin/TeamEvent/TeamEventDetails.tsx:164]: Hm, why are we calling a `fullReset` here? I know it's not fetching data, but that's what it feels like to the user when you hit the button. Can we write this so that we don't need to see the loading wheel when we hit the button?

**Comment by andrew032011** [comment]
Also, I'm slightly worried about the styling here. If this change is an urgent request from C&I lead feel free to add this as a fast follow-up PR afterwards.

In general, we're not too picky about styling for IDOL, but I feel that this one affects the UX a bit. Could we go for
- centering the cards for each section to their header (i.e. Members Approved, Members Rejected, Members Pending)
- make "Select Status" dropdown in-line with the left card in the "View all" view?

Screenshot 2024-02-28 at 10 40 53 PM

File: frontend/src/components/Admin/TeamEvent/TeamEventDetails.tsx
Comment (patriciaahuang) [reply frontend/src/components/Admin/TeamEvent/TeamEventDetails.tsx:164]: I agree that loading the entire page is unnecessary. Initially, `fullReset` was called to reset the dropdown to "Select Status" after clicking the "View All" button, but I found a way to do that without loading the entire page again.

**Comment by andrew032011** [comment]
@patriciaahuang it's not running because you have merge conflicts

**Review by andrew032011 — APPROVED** [review]
Styling looks good to me. I'd like to take one more look after you resolve merge conflicts.

**Review by andrew032011 — APPROVED** [review]

**Pull request merged**
